### ISSUE TITLE
git-clang-format suggested some odd indentation

### ISSUE THREAD

**Issue by tru** [llvm/llvm-project#67343]
This came up in https://github.com/llvm/llvm-project/pull/65744

Where `git-clang-format` suggested 4 spaces indentation on the patch. This also happened when I ran it locally:

```bash
gh pr checkout 65744
git-clang-format --diff b8b4ee6b450766796b162b4811a6b3f723d07268 d07486764071679c8cbcd5e7c4905eb41b4770b3 -- clang/utils/TableGen/ClangAttrEmitter.cpp
```

Just running `clang-format` on the code doesn't do this indentation, but it fixes a lot of other indentation problems. So I think `git-clang-format` picks up on some formatting outside the patch and misses the idea a bit.

Anyone have any idea what's going on here?

**Comment by owenca** [comment]
I couldn't reproduce it on either 17.0.1 or 18.0.0:
```
llvm-project$ gh pr checkout 65744
From https://github.com/llvm/llvm-project
 * [new ref]                   refs/pull/65744/head -> SourcePath
Switched to branch 'SourcePath'
llvm-project$ git-clang-format --diff b8b4ee6b450766796b162b4811a6b3f723d07268 d07486764071679c8cbcd5e7c4905eb41b4770b3 -- clang/utils/TableGen/ClangAttrEmitter.cpp
llvm-project$ clang-format -version
clang-format version 17.0.1
llvm-project$ git-clang-format --binary build_ninja/bin/clang-format --diff b8b4ee6b450766796b162b4811a6b3f723d07268 d07486764071679c8cbcd5e7c4905eb41b4770b3 -- clang/utils/TableGen/ClangAttrEmitter.cpp
llvm-project$ build_ninja/bin/clang-format -version
clang-format version 18.0.0 (https://github.com/llvm/llvm-project 7db211bda791e4326e3822220a57a882af20a273)
llvm-project$ 
```

**Comment by tru** [comment]
Confirmed that it's fixed in 17.0.1. I will update the clang-format version in our github action.
